src/test/java/org/broadinstitute/hellbender/tools/spark/sv/RunSGAViaProcessBuilderOnSparkUnitTest.java

```diff
@@ -0,0 +1,287 @@
+        for(final Integer d : editDistancesBetweenSeq){ Assert.assertEquals(d, zero); }
+
+        editDistancesBetweenSeq = new ArrayList<>();
+        final File duplicateRemovedFile = new File(workingDir, duplicateRemovedFileName);
+        final File actualMergedFile = new File(workingDir, RunSGAViaProcessBuilderOnSpark.runSGAFMMerge(sgaPath, duplicateRemovedFile, workingDir, threads, indexer, indexerArgs).fileNameToReturn);
+        final File expectedMergedFile = new File(TEST_DATA_DIR, filenamePrefix + ".pp.ec.filter.pass.rmdup.merged.fa");
+        final String mergedFileName = compareNamesAndComputeSeqEditDist(actualMergedFile, expectedMergedFile, false, editDistancesBetweenSeq);
+        for(final Integer d : editDistancesBetweenSeq){ Assert.assertEquals(d, zero); }
+
+        // final assembled contig test
+        final File mergedFile = new File(workingDir, mergedFileName);
+        final File actualAssembledContigsFile = new File(workingDir, RunSGAViaProcessBuilderOnSpark.runSGAOverlapAndAssemble(sgaPath, mergedFile, workingDir, threads).fileNameToReturn);
+        final RunSGAViaProcessBuilderOnSpark.ContigsCollection actualAssembledContigs = new RunSGAViaProcessBuilderOnSpark.ContigsCollection(actualAssembledContigsFile);
+        final RunSGAViaProcessBuilderOnSpark.ContigsCollection expectedAssembledContigs = new RunSGAViaProcessBuilderOnSpark.ContigsCollection(expectedAssembledContigFile);
+        Assert.assertTrue( compareContigs(actualAssembledContigs, expectedAssembledContigs) );
+    }
+
+    private static boolean compareContigs(final RunSGAViaProcessBuilderOnSpark.ContigsCollection actualAssembledContigs,
+                                          final RunSGAViaProcessBuilderOnSpark.ContigsCollection expectedAssembledContigs)
+            throws IOException, InterruptedException{
+
+        final SortedMap<Integer, String> expectedMap = collectContigsByLength(expectedAssembledContigs);
+
+        // first create mapping from read length to sequence
+        // (bad, but contig names are not guaranteed to be reproducible in different runs)
+        final SortedMap<Integer, String> actualMap = collectContigsByLength(actualAssembledContigs);
+
+
+        // then compare sequences, aware of RC
+        final SortedSet<Integer> actualLengthVacs = new TreeSet<>(actualMap.keySet());
+        final SortedSet<Integer> expectedLengthVals = new TreeSet<>(expectedMap.keySet());
+        Assert.assertEquals(actualLengthVacs, expectedLengthVals);
+
+        boolean result = true;
+        for(final Integer l : actualLengthVacs){
+
+            boolean sequencesAreTheSame = false;
+            boolean sequencesAreCloseEnough = false;
+
+            final String actualString = actualMap.get(l);
+            final String expectedString = expectedMap.get(l);
+            final String rcOfActualString = new String(BaseUtils.simpleReverseComplement(actualString.getBytes(StandardCharsets.UTF_8)));
```

Comment:
\[FindBugs\] WARNING: Dm: Found reliance on default encoding in org\.broadinstitute\.hellbender\.tools\.spark\.sv\.RunSGAViaProcessBuilderOnSparkUnitTest\.compareContigs\(RunSGAViaProcessBuilderOnSpark$ContigsCollection, RunSGAViaProcessBuilderOnSpark$ContigsCollection\): new String\(byte\[\]\)